modules/unsupported/wfs-ng/src/main/java/org/geotools/data/wfs/internal/WFSContentComplexFeatureCollection.java

```diff
@@ -72,7 +74,9 @@ public WFSContentComplexFeatureCollection(
 
         try {
             @SuppressWarnings("PMD.CloseResource") // wrapped and returned
-            InputStream stream = request.getFinalURL().openStream();
+            URL finalUrl = request.getFinalURL();
+            URLCheckerFactory.evaluate(finalUrl);
```

Comment:
So there is a data store parameter to control whether tests should be performed, but this test runs always instead? This is inconsistent, but also see other comments about this class.

Reply:
will add check